src/controller/java/templates/partials/decode_value.zapt

```diff
@@ -60,7 +60,7 @@ if ({{source}}.IsNull()) {
         env->SetByteArrayRegion({{target}}ByteArray, 0, static_cast<jsize>({{source}}.size()), reinterpret_cast<const jbyte *>({{source}}.data()));
         {{target}} = {{target}}ByteArray;
       {{else if (isCharString type)}}
-        {{target}} = env->NewStringUTF(std::string({{source}}.data(), {{source}}.size()).c_str());
+        chip::JniReferences::GetInstance().CharToStringUTF(std::string({{source}}.data(), {{source}}.size()).c_str(), {{target}});
```

Comment:
Shouldn't this just fail if the data is not valid UTF-8, instead of silently using a different value?

(Related: the code above this in the `isArray` does not fail on decoding errors, instead silently ignoring everything in the list after the decoding error....)

Reply:
In the case of String, if UTF-8 decoding fails, nullptr is returned. Is it correct to modify it so that it can return even the errorCode?

Reply:
e.g : 
CHIPAttributeTLVValueDecoder-src.zapt
`{{>decode_value target="value" source="cppValue" cluster=(asUpperCamelCase parent.name) depth=0 errorCode="*aError = " earlyReturn="nullptr"}}`

decode_value.zapt
`{{errorCode}} chip::JniReferences::GetInstance().CharToStringUTF(std::string({{source}}.data(), {{source}}.size()).c_str(), {{target}});`

Reply:
Something like that would make sense, yes, if that generates the right code.  Followup is fine for doing that as needed...

Reply:
I modified. Thanks. Please review it.